### DISCUSSION TITLE
Auto Homing overwrites the X, Y and Z position and sets it to 0?

### DISCUSSION THREAD

**Discussion by Allaboutzurich** [Jyers/Marlin#892] [Q&A] [answered]
I have the problem that when I "auto home" on my Creality Ender 3 v2 it sets the position of all axis to 0, even when the nozzle is in the middle of the bed. 

I tried several firmwares and configs and the same is happening with all of them. 
This means that I can only use my printer if I manually set the position to bottom left and then start a print. BLTouch 3.1 is not working because it will crash into the X max post. 

Has anyone an idea why this is happening or knows a way to always force the nozzle to bump into X min and Y min switch before overwriting the position?

**Comment by Tuncay-Ayhan** [comment]
> I have the problem that when I "auto home" on my Creality Ender 3 v2 it sets the position of all axis to 0, even when the nozzle is in the middle of the bed. 
> 
> 
> 
> I tried several firmwares and configs and the same is happening with all of them. 
> 
> This means that I can only use my printer if I manually set the position to bottom left and then start a print. BLTouch 3.1 is not working because it will crash into the X max post. 
> 
> 
> 
> Has anyone an idea why this is happening or knows a way to always force the nozzle to bump into X min and Y min switch before overwriting the position? 

You are probably using the wrong option to home. That one is meant for laser cutting if I remember it right. You need to "Home all axis"

**Reply by Allaboutzurich** [reply]
Thanks for you reply, would you mind guiding me to the corresponding section in the configuration so I can change it?

I searched for "home" and found the section for homing, but non of them sounded like they would do what you described. 

Then I thought why not looking up the section for G28 and thats where I found the "Quick Home" Section which has a code that overwrites the X and Y pos, but I do not feel comfortable to just comment it, what are your thoughts on this? 

Thanks a lot

**Comment by Tuncay-Ayhan** [comment]
Are you trying to home from the screen? Is that giving you problems? Also,
is your hardware setup stock?


Op ma 24 mei 2021 om 15:39 schreef Allaboutzurich ***@***.***>

**Reply by Allaboutzurich** [reply]
Hi Tuncay, I just tried to start the whole process with the Octoprint Terminal. 

My hardware is unmodified which means that everything I changed was, that I attached the BLTouch (with the BW Pins to the Z_Stop) and the Creality Filament Sensor where it belongs. Apart of that everything is like I bought it. I also order a 4.2.7 mainboard to make sure it is not because of a faulty mainboard. 

I recorded everything in a Log File and recorded it on video, I hope this helps to understand the situation a bit better :( 

Video Recording:
https://youtu.be/05SSVh4SJ74

Configurations:
[Marlin_Config.zip](https://github.com/Jyers/Marlin/files/6534543/Marlin_Config.zip)

Log File:
[Leveling.txt](https://github.com/Jyers/Marlin/files/6534536/Leveling.txt)

**Comment by Allaboutzurich** [comment, marked as answer]
I just looked trough the TXT Logfile and the first "unusual" behaviour started at line 385:

Recv: > X136.00 Y116.00 Z5.09
Recv: <<< do_blocking_move_to X136.00 Y116.00 Z5.09
Recv: current_position= X136.00 Y116.00 Z5.09 : Probe::set_deployed

The code sais correctly that it should move to X136 but instead it stayed on position X220.
The weird thing is, that the system thinks it is on position X136, what is obviously wrong.

The X stepper just stayed as it is instead of moving.

My findings so far. The logic question I would ask someone that showed me this behaviours is to ask if this situation happend befor while printing. No it did not. I can also manualy send the nozzle to X220 and back to X136 for example.

**Reply by Tuncay-Ayhan** [reply]
Is your probe offset correct? Because it looks like it thinks it can go a little more to the right off your bed but it can't

**Reply by Allaboutzurich** [reply]
That's what I thought too, and I will double check, but the problem is if I look into the log file, that it should move from testing point X2 Y1 (X220 Y116) to X1 Y1 (X136 Y116) and it thinks it moved but instead it stayed.

It then tested the same position 3 times and then moved to X2 Y2 (X220 Y220) instead of (X0 Y220). The next step is where it crashes into the post because it thinks it would move from testing point X0 Y2 (X0 Y220) to X1 Y2 (X136 Y220) but it is actually moving from X220 to X356 (220 + 136) which lays outside of the bed. 

The normal testing mode would be: 
Testing Point / Position
1. X0 Y0 / X0 Y0
2. X1 Y0 / X136 Y0
3. X2 Y0 / X220 Y0
4. X2 Y1 / X220 Y116
5. X1 Y1 / X136 Y116
6. X0 Y1 / X0 Y116
7. X0 Y2 / X0 Y220 
8. X1 Y2 / X136 Y220
9. X2 Y2 / X220 Y220

But you are the expert, I just wanted to give you my thoughts. I will therefore test the probe offset tomorrow when I have the chance. :) - Thanks a lot for your help!
